docs: Define OpenTelemetry Component Inclusion Principles (#6773)
grafana/alloy: docs/sources/reference/cli/otel.md

{{< param "PRODUCT_NAME" >}} {{< param "ALLOY_RELEASE" >}} bundles OpenTelemetry Collector components from version {{< param "OTEL_VERSION" >}}.
You can find more information about the bundled version in both the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector/tree/{{< param "OTEL_VERSION" >}}) and [OpenTelemetry Collector Contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/{{< param "OTEL_VERSION" >}}) repositories.

The {{< param "OTEL_ENGINE" >}} bundles a curated subset of these components rather than mirroring all of the OpenTelemetry Collector Contrib repository.

> {{< param "PRODUCT_NAME" >}} chooses each component based on demand and quality, maintains what it bundles, and keeps its dependency and security surface small and actively monitored.

Reviewer: ```suggestion
The bundled component set is based on demand, maturity, maintenance cost, dependency footprint, security risk, and license compatibility.
```

We can simp0ifly this a little

> Because the {{< param "OTEL_ENGINE" >}} is the Grafana distribution of the OpenTelemetry Collector, it always includes the components you need for a smooth integration with Grafana products.

Reviewer: ```suggestion
The {{< param "OTEL_ENGINE" >}} is the Grafana distribution of the OpenTelemetry Collector, and it includes components for common Grafana product integrations.
```

This turns the sentence around and avoids any over-promising. (If you start a sentence with "because" there's a good chance the sentence is backwards :-) )

Author: good point!

> For the criteria {{< param "PRODUCT_NAME" >}} uses to decide what to bundle, refer to the contributor guide on [adding OpenTelemetry components](https://github.com/grafana/alloy/blob/main/docs/developer/add-otel-component.md#inclusion-criteria).

Reviewer: ```suggestion
Refer to the [OpenTelemetry component contributor guide](https://github.com/grafana/alloy/blob/main/docs/developer/add-otel-component.md#inclusion-criteria) for information about the bundle criteria.
```
